### ISSUE TITLE
Recurring booking options

### ISSUE THREAD

**Issue by atlet** [Wunderbyte-GmbH/moodle-mod_booking#133]
When you create a new booking option, you can select if it's get created n times. You select interval (only available if start and end date are known - interval values are day, week, month and year) end how many records create. 
What about names? We call every option the same or we append record number? 
### Deleting
When you delete certain booking option you can select if delete only active or all recurring
### Database fields
In table booking_option we need to add new field parentid. With this field we will know, which booking options are connected.
### Editing
This option will be available only for creating new records. You can edit, but it wont change other options.

**Comment by dasistwas** [comment]
Still to clarify:

1. What fields are different in 1,2,3? Only the name? What will be added to the name of the booking option? They have to be unique.
2. Can you edit the nth booking option?
3. What happens when you edit the parent booking option?
4. Will the parent booking option be displayed as #1 booking option?

Idea: Wouldn't it be better to integrate that in the template system? And every recurring booking option has to be based on a template (approach 2). Have a look at: https://github.com/dasistwas/moodle-mod_booking/issues/132

**Comment by atlet** [comment]
> What fields are different in 1,2,3? Only the name? What will be added to the name of the booking option? They have to be unique.

I think it's ok if name is the same for all recurring booking options, as time of event is different.

> Can you edit the nth booking option?

Yes, but will not change other "connected" options.

> What happens when you edit the parent booking option?

Nothing, it will change only the edited option.

> Will the parent booking option be displayed as #1 booking option?

I think it make sense.

**Comment by dasistwas** [comment]
What is the essential feature of recurring booking option? I do not get it. You could also create n-copies of a template to create similar options. Also could you give feedback about integrating that in the templating feature instead of a separate "recurring" feature.

**Comment by atlet** [comment]
Yes, you can do also with templates, but it can  be slow. Let's say you have every week (on Monday at 10h) one event for 10 months. So you need to create 40 events. It's slow. If you have recurring events you just create 1 event and say, repeat every week for 10 months - system will create 40 events for you.

**Comment by dasistwas** [comment]
OK, but implementing it in templates would be:
1) Select template
2) Configure how often you create an option based on the template (Providing start date and interval)

**Comment by dasistwas** [comment]
So you could have a list with templates and for each template 2 buttons: 

- Create new booking option
- Create n new booking options

**Comment by atlet** [comment]
Yes, this option with templates it's a good proposition.
